interfaces/builtin/hugepages_control.go

```diff
@@ -0,0 +1,73 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt
+// https://www.kernel.org/doc/Documentation/vm/transhuge.txt
+// This interface assumes that huge pages are mounted at either:
+// - /dev/hugepages (Debian, Ubuntu)
+// - /run/hugepages (various documentation)
+const hugepagesControlSummary = `allows controlling hugepages`
+
+const hugepagesControlBaseDeclarationSlots = `
+  hugepages-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const hugepagesControlConnectedPlugAppArmor = `
+# Allow configuring huge pages via /sys or /proc
+/sys/kernel/mm/hugepages/{,hugepages-[0-9]*}/* r,
```

Comment:
My Raspberry Pi running core16 does not seem to have this hierarchy. Is that expected? Should the connected interface insert a kernel module?

Reply:
You need to either insert some kernel parameters, for instance:
`default_hugepagesz=2097152 hugepagesz=2097152 hugepages=1024`
or mount a hugetblfs:
`mount -t hugetlbfs -o pagesize=2M nodev /dev/hugepages`
No kernel module needed here.